jax/_src/scipy/special.py

```diff
@@ -657,3 +661,290 @@ def i1e(x):
+      points are nearly equal-spaced along θ and provide exact discrete
+      orthogonality, (P^m)^T W P_m = I, where `T` represents the transpose
+      operation, `W` is a diagonal matrix containing the quadrature weights,
+      and `I` is the identity matrix. The Gauss-Chebyshev points are equally
+      spaced, which only provide approximate discrete orthogonality. The
+      Driscoll & Healy qudarture points are equally spaced and provide the
+      exact discrete orthogonality. The number of sampling points is required to
+      be twice as the number of frequency points (modes) in the Driscoll & Healy
+      approach, which enables FFT and achieves a fast spherical harmonics
+      transform.
+    is_normalized: True if the associated Legendre functions are normalized.
+      With normalization, `N_l^m` is applied such that the spherical harmonics
+      form a set of orthonormal basis functions of L^2(S^2).
+
+  Returns:
+    The 3D array of shape `(l_max + 1, l_max + 1, len(x))` containing the values
+    of the ALFs at `x`; the dimensions in the sequence of order, degree, and
+    evalution points.
+  """
+  p = jnp.zeros((l_max + 1, l_max + 1, x.shape[0]))
+
+  a_idx = jnp.arange(1, l_max + 1)
+  b_idx = jnp.arange(l_max)
+  if is_normalized:
+    initial_value = 0.5 / jnp.sqrt(jnp.pi)  # The initial value p(0,0).
+    f_a = jnp.cumprod(-1 * jnp.sqrt(1.0 + 0.5 / a_idx))
+    f_b = jnp.sqrt(2.0 * b_idx + 3.0)
+  else:
+    initial_value = 1.0  # The initial value p(0,0).
+    f_a = jnp.cumprod(1.0 - 2.0 * a_idx)
+    f_b = 2.0 * b_idx + 1.0
+
+  p = p.at[(0, 0)].set(initial_value)
+
+  # Compute the diagonal entries p(l,l) with recurrence.
+  y = jnp.cumprod(
+      jnp.broadcast_to(jnp.sqrt(1.0 - x * x), (l_max, x.shape[0])),
+      axis=0)
+  p_diag = initial_value * jnp.einsum('i,ij->ij', f_a, y)
+  diag_indices = jnp.diag_indices(l_max + 1)
+  p = p.at[(diag_indices[0][1:], diag_indices[1][1:])].set(p_diag)
+
+  # Compute the off-diagonal entries with recurrence.
+  p_offdiag = jnp.einsum('ij,ij->ij',
+                         jnp.einsum('i,j->ij', f_b, x),
+                         p[jnp.diag_indices(l_max)])
+  offdiag_indices = (diag_indices[0][:l_max], diag_indices[1][:l_max] + 1)
+  p = p.at[offdiag_indices].set(p_offdiag)
+
+  # Compute the remaining entries with recurrence.
+  d0_mask_3d, d1_mask_3d = _gen_recurrence_mask(
+      l_max, is_normalized=is_normalized)
+
+  def body_fun(i, p_val):
+    coeff_0 = d0_mask_3d[i]
+    coeff_1 = d1_mask_3d[i]
+    h = (jnp.einsum('ij,ijk->ijk',
+                    coeff_0,
+                    jnp.einsum(
+                        'ijk,k->ijk', jnp.roll(p_val, shift=1, axis=1), x)) -
+         jnp.einsum('ij,ijk->ijk', coeff_1, jnp.roll(p_val, shift=2, axis=1)))
+    p_val = p_val + h
+    return p_val
+
+  p = lax.fori_loop(lower=2, upper=l_max+1, body_fun=body_fun, init_val=p)
+
+  return p
+
+
+@_wraps(osp_special.lpmn)
```

Comment:
Note that using `_wraps` will overwrite the function's docstring with a modified version of the docstring from the wrapped function.

We generally use `_wraps()` for convenience, along with an optional `lax_description` argument to note departures from the wrapped API.

If you'd prefer to keep your docstring, you can remove the `_wraps` decorator.

Reply:
Thanks for the explanations. Removed decorator `_wraps`.